src/libcore/ops.rs

```diff
@@ -95,6 +95,16 @@ use fmt;
 #[stable(feature = "rust1", since = "1.0.0")]
 pub trait Drop {
     /// A method called when the value goes out of scope.
+    ///
+    /// When this method has been called, `self` has not yet been deallocated.
+    /// If it were, `self` would be a dangling reference.
+    ///
+    /// After this function is over, the memory of `self` will be deallocated.
```

Comment:
Maybe instead of 'deallocated', 'deinitialized' would be more accurate? It could even say, 'deinitialized, and if `self` is heap-allocated, deallocated', or 'and possibly deallocated'.